simulator/include/icache.h

```diff
@@ -0,0 +1,72 @@
+#ifndef CHS_I_CACHE_H
```

Comment:
Should we rename this class to something like ```CacheInterface```? Don't want to get it confused with a class representing an instruction cache.

Reply:
I have updated the interface convention from IName to NameInterface